ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -299,17 +298,7 @@ private void runReAct(
         String maxIteration = Optional.ofNullable(tmpParameters.get("max_iteration")).orElse("3");
 
         // Create root interaction.
-        StepListener<CreateInteractionResponse> createRootItListener = new StepListener<>();
```

Comment:
I think there is a race condition between root interaction and child interaction if we remove this listener. If child interaction creation is before root interaction creation, we can't get parent interaction id for child interaction.

Reply:
AgentExecutor invokes Chat agent  only after parent interaction ID is created. https://github.com/opensearch-project/ml-commons/pull/1707/files#diff-31577db9d5543dc8d9e7a8aa242aa2b046cded8685c2f19855f3ae3211ffb8e1R136-R144

Parent interaction will be present when child interactions are saved.

Reply:
Got it, thanks!